src/components/connect.js

```diff
@@ -45,11 +46,12 @@ export default function connect(mapStateToProps, mapDispatchToProps, mergeProps,
   // Helps track hot reloading.
   const version = nextVersion++
 
-  function computeMergedProps(stateProps, dispatchProps, parentProps) {
+  function computeMergedProps(componentDisplayName, stateProps, dispatchProps, parentProps) {
     const mergedProps = finalMergeProps(stateProps, dispatchProps, parentProps)
     invariant(
```

Comment:
We should refactor so that this check also occurs inside `checkStateShape()`. I would pass `methodName` (e.g. `'mapStateToProps'`, `mapDispatchToProps'` or `'mergeProps'`) instead of the existing boolean `dispatch` argument.